hoot-core/src/main/cpp/hoot/core/conflate/polygon/BuildingMerger.cpp

```diff
@@ -437,6 +474,142 @@ std::shared_ptr<Element> BuildingMerger::buildBuilding(const OsmMapPtr& map,
   }
 }
 
+RelationPtr BuildingMerger::combineConstituentBuildingsIntoRelation(
+  const OsmMapPtr& map, std::vector<ElementPtr>& constituentBuildings, const bool preserveTypes)
+{
+  if (constituentBuildings.size() == 0)
+  {
+    throw IllegalArgumentException("No constituent buildings passed to building merger.");
+  }
+
+  // This is primarily put here to support testable output.
+  InMemoryElementSorter::sort(constituentBuildings);
+
+  // Just looking for any key that denotes multi-level buildings. Not handling the situation where
+  // a non-3D building is merging with a 3D building...not exactly sure what to do there...create
+  // both a multipoly and building relation (even though it wouldn't be valid)? No point in worrying
+  // about it until its seen in the wild.
+  QStringList threeDBuildingKeys;
+  threeDBuildingKeys.append(MetadataTags::BuildingLevels());
+  threeDBuildingKeys.append(MetadataTags::BuildingHeight());
+  const bool allAreBuildingParts =
```

Comment:
Debatable what behavior should happen if some, but not all, of the buildings have 3D tags... Technically, I think that is incorrect input. So, I'm just logging a warning and skipping adding the building relation (adding multipoly instead)...open to other suggestions.

Reply:
I think this is the best course of action given that it would have to be bad input data to cause that situation so it should be reported.